File: backend/check_user_table.py
#!/usr/bin/env python3
"""
检查用户表的实际结构
"""

import os
import psycopg2
from dotenv import load_dotenv

# 加载环境变量
load_dotenv()

def check_user_table():
    """检查用户表结构"""
    try:
        # 从环境变量获取数据库连接信息
        db_url = os.getenv("DATABASE_URL")
        if not db_url:
            print("错误: 未找到DATABASE_URL环境变量")
            return False
            
        # 连接数据库
        conn = psycopg2.connect(db_url)
        cursor = conn.cursor()
        
        # 查询用户表结构
        cursor.execute("""
        SELECT column_name, data_type, is_nullable, column_default
        FROM information_schema.columns 
        WHERE table_name = 'user' 
        ORDER BY ordinal_position;
        """)
        
        columns = cursor.fetchall()
        
        print("用户表结构:")
        print("-" * 60)
        for col in columns:
            print(f"{col[0]:<20} {col[1]:<15} {col[2]:<10} {col[3] or ''}")
        
        cursor.close()
        conn.close()
        return True
        
    except Exception as e:
        print(f"❌ 检查失败: {e}")
        return False

if __name__ == "__main__":
    check_user_table()

File: backend/check_user_table.js
import { Pool } from 'pg';

const pool = new Pool({
  host: 'localhost',
  port: 5432,
  database: 'admagic',
  user: 'postgres',
  password: '123456'
});

async function checkUserTable() {
  try {
    const result = await pool.query(`
      SELECT column_name, data_type, is_nullable, column_default
      FROM information_schema.columns 
      WHERE table_name = 'user' 
      ORDER BY ordinal_position;
    `);
    
    console.log('用户表结构:');
    console.table(result.rows);
    
    // 检查是否有数据
    const countResult = await pool.query('SELECT COUNT(*) FROM "user"');
    console.log('用户表记录数:', countResult.rows[0].count);
    
    // 检查其他Better Auth相关表
    const tables = ['session', 'account', 'verification'];
    for (const table of tables) {
      try {
        const tableResult = await pool.query(`
          SELECT column_name, data_type 
          FROM information_schema.columns 
          WHERE table_name = $1 
          ORDER BY ordinal_position;
        `, [table]);
        
        console.log(`\n${table}表结构:`);
        console.table(tableResult.rows);
      } catch (err) {
        console.log(`${table}表不存在或查询失败:`, err.message);
      }
    }
    
  } catch (error) {
    console.error('数据库查询错误:', error.message);
  } finally {
    await pool.end();
  }
}

checkUserTable();